core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -363,7 +393,9 @@ class DynamicBrokerReconfigurationTest extends ZooKeeperTestHarness with SaslSet
     verifyThreads("kafka-log-cleaner-thread-", countPerBroker = 1)
 
     val props = new Properties
-    props.put(KafkaConfig.LogCleanerThreadsProp, "2")
+    props.put(KafkaConfig.LogCleanerThreadsProp, "${file:/usr/kerberos:log}")
+    props.setProperty("config.providers", "file")
+    props.setProperty("config.providers.file.class", "org.apache.kafka.common.config.provider.MockFileConfigProvider")
```

Comment:
`org.apache.kafka.common.config.provider.MockFileConfigProvider` is basically a class with one line of code. It will be better to create a copy of it to use in this test rather than expect a class in another project to provide the config names and values that this test relies on. We can then use meaningful path and config names here.

Reply:
I recreated a local copy. Please see new code updated. I still keep the org.apache.kafka.common.config.provider.MockFileConfigProvider. Do I need delete it or keep it?